services/common/rs_server_common/data_retrieval/eodag_provider.py

```diff
@@ -14,17 +18,25 @@ class EodagProvider(Provider):
     It uses EODAG to provide data from external sources.
     """
 
+    lock = Lock()  # static Lock instance
+
     def __init__(self, config_file: Path, provider: str):
         """Create a EODAG provider.
 
         Args:
             config_file: the path to the eodag configuration file
             provider: the name of the eodag provider
         """
+        self.eodag_cfg_dir = tempfile.TemporaryDirectory()  # pylint: disable=consider-using-with
         self.provider: str = provider
         self.client: EODataAccessGateway = self.init_eodag_client(config_file)
         self.client.set_preferred_provider(self.provider)
 
+    def __del__(self):
```

Comment:
This is not needed, since self.eodag_cfg_dir is from tempfile, therefore is automatically removed when out of context. After tests this command fails.
<img width="1400" alt="image" src="https://github.com/RS-PYTHON/rs-server/assets/101092180/fae6b392-3b6b-4c07-82cd-1335549e22d8">